docs/getting-started/publish-infrastructure.md

```diff
@@ -410,6 +413,8 @@ spec:
         - fromConnectionSecretKey: username
         - fromConnectionSecretKey: password
         - fromConnectionSecretKey: endpoint
+        - name: port
+          value: NTQzMg==  # The string 5432 base64 encoded.
```

Comment:
It's arguably a little odd to specify these values here next to the composed resource, since they really just bubble up to the composite resource regardless of which composed resource they came from. I figured it was nice to add it here for two reasons:

* It's nice to conceptually model which composed resource an injected key is associated with - i.e. in this case the ports are for the underlying PostgreSQL instances, not the supporting infrastructure.
* It was doable without a breaking API change.

Reply:
I suspect the last key wins.

The logic here probably needs some attention. We may want to raise an error if a key is specified twice. Also, I believe the InfraDef's set of connection secret keys are currently used as a filter, to ensure _at most_ those keys bubble up from the composed resources. I'd prefer if it was more of a contract, requiring that each of its specified keys was supplied exactly once by a composed resource.

Reply:
Agreed, I think that this can be addressed in a follow-up and we can move ahead with this PR :+1: